### ISSUE TITLE
Support getDirectory() for non-default bucket

### ISSUE THREAD

**Issue by a-sully** [whatwg/fs#112]
See https://wicg.github.io/storage-buckets/#issue-25ffaa8c

The issue mentioned in the storage buckets spec is that we need a version of [this algorithm](https://whatpr.org/fs/110.html#dom-storagemanager-getdirectory) which works for any bucket. You'll notice if you drill down into "obtain a local storage bottle map" that it currently assumes the "default" bucket. This just means passing an additional parameter.

We do not currently have a way to go from bottle to bucket for any storage endpoint.

_Originally posted by @evanstade in https://github.com/whatwg/fs/pull/110#discussion_r1160766995_

**Comment by a-sully** [comment]
https://github.com/whatwg/storage/issues/2 is the meta-bug for making non-default buckets work, and it seems like addressing that is a prerequisite for adding another parameter to "obtain a local storage bottle map" to make it work for non-default buckets

**Comment by evanstade** [comment]
Adding another parameter to "obtain a local storage bottle map" is (part of) what addresses that metabug, no?

**Comment by a-sully** [comment]
Correct. Apologies, that was poor wording on my part. My understanding of the steps are:
- Decide how to make non-default buckets work https://github.com/whatwg/storage/issues/2
- Specify how non-default buckets will work:
  - Add another parameter to "obtain a local storage bottle map" to make it work for non-default buckets
  - etc...
- Update this spec (and other storage specs) to take into account non-default buckets:
  - Use the new "obtain a local storage bottle map" algorithm in getDirectory()
  - etc...
